addons/mail/static/src/messaging/entity/notification_group/notification_group.js

```diff
@@ -0,0 +1,109 @@
+odoo.define('mail.messaging.entity.NotificationGroup', function (require) {
+'use strict';
+
+const { registerNewEntity } = require('mail.messaging.entityCore');
+const { attr, one2many } = require('mail.messaging.EntityField');
+
+function NotificationGroupFactory({ Entity }) {
+
+    class NotificationGroup extends Entity {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * Opens the view that allows to cancel all notifications of the group.
+         */
+        openCancelAction() {
+            const device = this.env.messaging.device;
+            if (this.notification_type === 'email') {
+                this.env.do_action('mail.mail_resend_cancel_action', {
+                    additional_context: {
+                        default_model: this.model,
+                        unread_counter: this.notifications.length,
+                    },
+                });
+            }
+            if (!device.isMobile) {
+                this.env.messaging.messagingMenu.close();
+            }
+        }
+
+        /**
+         * Opens the view that displays either the single record of the group or
+         * all the records in the group.
+         */
+        openDocuments() {
+            if (this.res_id) {
+                if (this.model === 'mail.channel') {
+                    const channel = this.env.entities.Thread.insert({
+                        id: this.res_id,
+                        model: this.model,
+                    });
+                    channel.open();
+                } else {
+                    this.env.messaging.openDocument({
+                        id: this.res_id,
+                        model: this.model,
```

Comment:
would replace `this.model` by `mail.channel`. I know this require 3 changes of LOC if we ever have to change model of channels, but on one hand it's one less variable to care and bare string is much easier to read than a variable. What do you think?

Reply:
I think you meant to comment the one inside the `if (this.model === 'mail.channel')` and not the `else`.

I don't really care either way, I guess this would make sense if we had channel entity, we would do "entity channel open" and just give it the id and not the model.

Reply:
But it's not one less variable though, it still relies on the use of this.model on the if itself, and if the code is moved outside of that condition it will break/not make sense.

So I guess it is safer to keep it like it is for now?